docs/how-to/detect-ipfs-on-web.md

```diff
@@ -0,0 +1,127 @@
+# Detect IPFS on the Web
+
+Below is the suggested heuristic for detecting content-addressed resources on the web, and presenting additional user interface for interacting with them.
+
+The goal of this page is to provide some suggestions and best practices for web browsers and other [user agents](https://en.wikipedia.org/wiki/User_agent) that are interested in adding support for [content-addressed resources on the Web](/how-to/address-ipfs-on-web).
+
+## Detecting content-addressed resources
+
+In many cases IPFS resource can be detected just by looking at URI, especially if it is a HTTP URL pointing at a public gateway. This allows browser to upgrade transport protocol to IPFS, and removed the need for sending HTTP request to a remote server.
+
+The high level heuristic is to check, in order:
+- Does the `URL` follow [path or subdomain gateway](/how-to/address-ipfs-on-web/) convention?
+- If not, does [DNSLink](/concepts/dnslink/) with an IPFS path exist for `URL.hostname`?
+
+If any of the above is true, an user agent should be able to find a content path that can be used for data retrieval over IPFS. See the illustration below.
+
+```mermaid
+flowchart TD
+    Start[Top-level navigation event]
```

Comment:
If diagram only handles top-level navigation events, should add a note because policy there is typically different than other load points. Maybe point also to the test page.

Reply:
The path detection diagram is generic enough to be applied to subresource requests too – the usual limitations we experience are vendor specific (example: https://github.com/ipfs/ipfs-companion/issues/1095).

TODO(note to self): think how to convey that. Probably change labels on diagram to be about all requests + add section explaining  caveats around "top-level navigaton" and "subresource requests" on HTTP and `ipfs://`.